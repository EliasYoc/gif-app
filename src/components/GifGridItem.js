import React from "react";
import PropTypes from "prop-types";
export const GifGridItem = ({ img, title, id }) => {
  return (
    <article className="card animate__animated animate__fadeInLeft">
      <h3>{title}</h3>
      <img
        style={{ flexGrow: 1, objectFit: "contain" }}
        loading="lazy"
        src={img}
        alt={title}
      />
    </article>
  );
};

GifGridItem.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
