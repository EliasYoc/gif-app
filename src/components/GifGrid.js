import React from "react";
import useFetchGif from "../hooks/useFetchGif";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGif(category);
  return (
    <>
      <h3>{category}</h3>
      <section className="card-grid">
        {loading && <h5>Cargando...</h5>}
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </section>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
