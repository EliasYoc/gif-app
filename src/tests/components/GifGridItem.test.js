import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas para <GifGridItem/>", () => {
  const url =
    "https://giphy.com/gifs/TOEIAnimationUK-goku-dragon-ball-super-clnwbP8SbJUvb0Nxty";
  const title = "titulo";
  const wrapper = shallow(<GifGridItem img={url} title={title} />);

  test("debe ejecutar <GifGridItem/> correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe tener un parrafo con texto", () => {
    const h3 = wrapper.find("h3");
    expect(h3.text().trim()).toBe(title);
  });
  test("que la img tenga src y alt", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
  test("img debe tener la class animate__fadeInLeft", () => {
    const article = wrapper.find("article");
    const className = "animate__fadeInLeft";

    expect(article.prop("className").includes(className)).toBe(true);
  });
});
