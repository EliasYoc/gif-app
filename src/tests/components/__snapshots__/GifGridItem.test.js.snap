// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGridItem/> debe ejecutar <GifGridItem/> correctamente 1`] = `
<article
  className="card animate__animated animate__fadeInLeft"
>
  <h3>
    titulo
  </h3>
  <img
    alt="titulo"
    loading="lazy"
    src="https://giphy.com/gifs/TOEIAnimationUK-goku-dragon-ball-super-clnwbP8SbJUvb0Nxty"
    style={
      Object {
        "flexGrow": 1,
        "objectFit": "contain",
      }
    }
  />
</article>
`;
